chaossearch/client/operation_read_indexing_state.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// For documentation see: https://docs.chaossearch.io/reference#bucketmodel
+func (client *Client) ReadIndexingState(ctx context.Context, req *ReadIndexingStateRequest) (*IndexingState, error) {
+	method := "GET"
+
+	responseBody, err := makeGetBucketMetadataRequest(method, client, ctx)
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %s", err)
+	}
+
+	var response ReadBucketMetadataResponse
```

Comment:
Should the `ReadBucketMetadataResponse` struct really be public if the `makeGetBucketMetadataRequest` method is not?

Reply:
I was thinking that maybe in the future we'll extract this method to the client or smth like this. And this struct is not specific to this operation only, so it makes sense for it to be public. But I don't have a strong opinion on it and can change it if you want so

Reply:
To not wait for a seaparate discussion I just did as you suggested :)

Reply:
Either is fine with me. Thanks for addressing